x-pack/platform/plugins/shared/stack_alerts/public/rule_types/es_query/expression/esql_query_expression.tsx

```diff
@@ -45,6 +47,21 @@ import {
   ALERT_ID_SUGGESTED_MAX,
 } from '../../../../common';
 
+const getTimeFilter = (
+  queryService: DataPublicPluginStart['query'],
+  uiSettings: IUiSettingsClient,
+  timeFieldName?: string
+) => {
+  const esQueryConfigs = getEsQueryConfig(uiSettings);
+  const timeFilter =
+    queryService.timefilter.timefilter.getTime() &&
+    getTime(undefined, queryService.timefilter.timefilter.getTime(), {
+      fieldName: timeFieldName,
+    });
+
+  return buildEsQuery(undefined, [], [...(timeFilter ? [timeFilter] : [])], esQueryConfigs);
```

Comment:
I think there's no need to spread here:


```suggestion
  return buildEsQuery(undefined, [], timeFilter ? [timeFilter] : [], esQueryConfigs);
```